### ISSUE TITLE
Problems with web identity token (OIDC) in EKS cluster

### ISSUE THREAD

**Issue by Mistra** [fluent/fluent-bit#8278]
## Bug Report

**Describe the bug**
On EKS, Fluentbit is not able to assume the service account provided role.

**To Reproduce**
Annotate a service account with proper role and make sure that fuentbit's annotated pod has both AWS_WEB_IDENTITY_TOKEN_FILE env variable and the file pointed in place. Also, SA is annotated with the proper tag role_arn arn:aws:iam::1111111111111:role/cloudwatch-log-event-writer
- Example log message:
```
<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15">
  <Error>
    <Type>Sender</Type>
    <Code>
      AccessDenied</Code>
    <Message>User:
      arn:aws:sts::1111111111111:assumed-role/cloudwatch-log-event-writer/AAAAAAAAAAAAAAAAAAA
      is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::1111111111111:role:cloudwatch-log-event-writer</Message>
  </Error>
  <RequestId>...</RequestId>
</ErrorResponse>
```
- Steps to reproduce the problem:

**Expected behavior**

Beeing able to assume the role specified (and therefore, push logs to cloudwatch). To test if the environment was ok, I mounted the debug image of fluentbit, connected inside the running container in a shell, installed aws cli v2 and managed to correctly list all log groups of Cloudwatch after assuming the role.

**Your Environment**

* Version used: fluentbit v 2.2.0, installed with fluent-bit helm chart at https://fluent.github.io/helm-charts, v. 0.40.0
* Configuration:
  outputs: |
    [OUTPUT]
        Name cloudwatch_logs
        Match *
        region eu-west-1
        log_group_name /aws/eks/dev/dev-eks-cluster/sandbox-api
        role_arn arn:aws:iam::1111111111111:role:cloudwatch-log-event-writer
        log_stream_prefix from-fluent-bit-
        auto_create_group true
* Environment name and version (e.g. Kubernetes? What version?): EKS v. 1.28
* Filters and plugins:
Cloudwatch OUTPUT plugin

**Additional context**
I should end up using case 3 as stated [here](https://github.com/fluent/fluent-bit-docs/blob/master/administration/aws-credentials.md). I saw no additional documentation of particular parameters to use in the configurations.

**Comment by celalettin1286** [comment]
Hello @Mistra 

As you mentioned [here](https://github.com/fluent/fluent-bit-docs/blob/master/administration/aws-credentials.md), EKS Web Identity Token (OIDC) fetches credentials via a signed web identity token for a Kubernetes service account. 

It means you don't need any explicit configuration(role_arn arn:aws:iam::1111111111111:role:cloudwatch-log-event-writer). 

you may see the steps [here](https://anywhere.eks.amazonaws.com/docs/clustermgmt/observability/fluentbit-logging/) and AFAIU all steps already applied and only thing required is to remove role_arn from the config. Could you please try to remove role_arn(required for cross account access as stated in the doc) and let us know the result?

**Comment by Mistra** [comment]
Thank you for the quick answer.
In the end I messed with the helm chart parameters, it was a misconfiguration at chart level.
Sorry for bothering. Another thing though, I noticed the configuration you pointed me at was using a different image than the one you release.
`public.ecr.aws/aws-observability/aws-for-fluent-bit:stable` to be exact. I swapped them and I saw for my purposes are interchangeable.
Thank you again for the help.
